lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+
+    def add_file(self, file_name: str, label: str, content: str):
+        """Adds a file to the corpus
+
+        :param content: file content
+        :param file_name: file name
+        :param label: file label
+        """
+        total_word_counts = len(str(content).split(" "))
+        self._corpus.append(File(content=content,
+                                 file_name=file_name,
+                                 label=label,
+                                 total_word_counts=total_word_counts))
+
+    def add_dictionary(self, file_name: str, label: str, content: str):
+        """Adds a dictionary
+
+        :param file_name: name of the file
+        :param label: label of the file
+        :param content: content of the file
+        """
+        new_list = str(content).split(", ")
+        new_list.sort(key=lambda x: len(x.split()), reverse=True)
+        self._dictionaries.append(Dictionary(content=new_list,
+                                             file_name=file_name,
+                                             label=label))
+
+    def get_active_dicts(self) -> list:
+        """
+
+        :return: a list containing all active dictionaries
+        """
+        return [dictionary for dictionary in self.dictionaries
+                if dictionary.active]
+
+    def count(self):
+        """counts all dictionaries for all active files in the corpus"""
+        self._counters = []
+        dictionaries = self.join_active_dicts()
+        for file in deepcopy(self._corpus):
+            dictionaries = count_phrases(dictionaries, file)
+            self.get_dictionary_counts(dictionaries)
+
+    def get_dictionary_counts(self, dictionaries):
+        """gets the counts for each dictionary
+
+        :param dictionaries: list of Phrase object
+        """
+        counter = []
+        active_dicts = self.get_active_dicts()
+        for dictionary in active_dicts:
+            count = 0
+            for phrase in dictionaries:
+                if phrase.dict_label == dictionary.label:
+                    count += phrase.count
+            counter.append(count)
+            if len(counter) == len(active_dicts):
+                self._counters.append(counter)
+
+    def generate_scores(self):
+        """calculate the formula and scores=formula/total_word_count for each
+        file in the corpus
+
+        """
+        self._scores = []
+        self._formulas = []
+        active_dicts = self.get_active_dicts()
+        result = 0
+        for i in range(len(self._corpus)):
+            new_formula = self._formula
+            for j in range(len(active_dicts)):
+                new_formula = new_formula.replace(
+                    "[" + active_dicts[j].label + "]",
+                    str(self._counters[i][j]))
+            new_formula = new_formula.replace("()", "")
+            try:
+                result = eval(new_formula)
+            except (ValueError, SyntaxError):
+                pass
```

Comment:
It's bad to let errors pass silently. Does the user even know their formula did nothing?

Reply:
my bad, this try/except is not needed anymore

Reply:
So, you can guarantee that the user will not be able to input a invalid equation?